src/redisHashKeyValueDB.ts

```diff
@@ -0,0 +1,159 @@
+import {
+  CommonKeyValueDBSaveBatchOptions,
+  CommonDBCreateOptions,
+  CommonKeyValueDB,
+  KeyValueDBTuple,
+} from '@naturalcycles/db-lib'
+import { _chunk, _mapObject, StringMap } from '@naturalcycles/js-lib'
+import { ReadableTyped } from '@naturalcycles/nodejs-lib'
+import { RedisClient } from './redisClient'
+import { RedisKeyValueDBCfg } from './redisKeyValueDB'
+
+export interface RedisHashKeyValueDBCfg extends RedisKeyValueDBCfg {
+  hashKey: string
+}
+
+export class RedisHashKeyValueDB implements CommonKeyValueDB, AsyncDisposable {
+  client: RedisClient
+  keyOfHashField: string
+
+  constructor(cfg: RedisHashKeyValueDBCfg) {
+    this.client = cfg.client
+    this.keyOfHashField = cfg.hashKey
+  }
+
+  async ping(): Promise<void> {
+    await this.client.ping()
+  }
+
+  async [Symbol.asyncDispose](): Promise<void> {
+    await this.client.disconnect()
+  }
+
+  async getByIds(table: string, ids: string[]): Promise<KeyValueDBTuple[]> {
+    if (!ids.length) return []
+    // we assume that the order of returned values is the same as order of input ids
+    const bufs = await this.client.hmgetBuffer(this.keyOfHashField, this.idsToKeys(table, ids))
+    return bufs.map((buf, i) => [ids[i], buf] as KeyValueDBTuple).filter(([_k, v]) => v !== null)
+  }
+
+  async deleteByIds(table: string, ids: string[]): Promise<void> {
+    if (!ids.length) return
+    await this.client.hdel(this.keyOfHashField, this.idsToKeys(table, ids))
+  }
+
+  async saveBatch(
+    table: string,
+    entries: KeyValueDBTuple[],
+    opt?: CommonKeyValueDBSaveBatchOptions,
+  ): Promise<void> {
+    if (!entries.length) return
+
+    const map: StringMap<any> = _mapObject(Object.fromEntries(entries), (k, v) => [
```

Comment:
p3: I'd use native map here (instead of _mapObject), like:

```
const map = Object.fromEntries(entries.map(...))
```

why?
Since you're already using native Object.fromEntries.